test/tests/pbs_Rrecord_resources_used.py

```diff
@@ -0,0 +1,616 @@
+            self.server.expect(JOB, {ATTR_state: 'F'}, jid2, extend='x')
+            self.server.expect(JOB, {ATTR_state: 'Q'}, jid3s1)
+
+        """
+        tracejob should show "Job requeued, execution node <node name> down"
+        """
+        m = self.server.tracejob_match(
+        msg='Job requeued, execution node .* down', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+
+        if is_nonrerunnable==False:
+            m = self.server.tracejob_match(
+            msg='Job requeued, execution node .* down', id=jid2, regexp=True)
+            self.assertNotEqual(m, None)
+        else:
+            m = self.server.tracejob_match(
+            msg='Job requeued, execution node .* down', id=jid2, regexp=True)
+            self.assertIsNone(m)
+
+        m = self.server.tracejob_match(
+        msg='Job requeued, execution node .* down', id=jid3s1, regexp=True)
+        self.assertNotEqual(m, None)
+
+        m = self.server.accounting_match(
+        msg='.*Resource_List.*', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*Resource_List.*', id=jid2, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*Resource_List.*', id=jid3s1, regexp=True)
+        self.assertNotEqual(m, None)
+
+        if restart_mom == 's':
+            """
+            Start mom without any  option
+            """
+            self.momA.start()
+            self.momB.start()
+        elif restart_mom == 'r':
+            """
+            Start mom with -r option
+            """
+            self.momA.start(args=['-r'])
+            self.momB.start(args=['-r'])
+        else:
+            print "Mom is not running\n"
+
+        return  jid1,jid2,jid3s1
+
+    def test_t1(self):
+        """
+        Scenario: The node on which the job was running goes down and node_fail_requeue time-out is hit. 
+        Expected outcome: Server should record last known resource usage in the 'R' record.
+        """
+        jid1,jid2,jid3s1 = self.common(False, False)
+
+        m = self.server.accounting_match(
+        msg='.*R;' + jid1 + '.*resources_used.*', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*R;' + jid2 + '.*resources_used.*', id=jid2, regexp=True)
+        self.assertNotEqual(m, None)
+        # Use re.replace() as accounting_match cannot match subjobs []
+        m = self.server.accounting_match(
+        msg='.*R;' + re.escape(jid3s1) + '.*resources_used.*', id=jid3s1, regexp=True)
+        self.assertNotEqual(m, None)
+
+    def test_t2(self):
+        """
+        Scenario: The node on which the job was running goes down and node_fail_requeue time-out is hit 
+                  and mom is restarted with '-r'
+        Expected outcome: Server should record last known resource usage in the 'R' record.
```

Comment:
You're doing more in this test than what is listed here.  Maybe make the docstring a bit more complete?